zp-relayer/endpoints.ts

```diff
@@ -146,8 +146,13 @@ async function getJob(req: Request, res: Response) {
     const poolJobState = await job.getState()
     if (poolJobState === 'completed') {
       // Transaction was included in optimistic state, waiting to be mined
+      if (job.returnvalue === null) {
+        job = await poolTxQueue.getJob(jobId)
+        // Sanity check
+        if (!job) throw new Error('Pool job not found')
+    }
```

Comment:
You fetch job again, but `returnvalue` property of this object is still unchecked. Imagine you have another issue reason (not concurrent nature) which leads `job.returnvalue` equals null (even after second job request). I think here should be some kind of limited-iteration while cycle, or secondary checking `job.returnvalue` with proper internal error returning

Reply:
If we got `completed` job's state there should be no reasons why `returnvalue` could be `null` (possible only if the job returned `null` itself, but there is no such cases in current implementation). I think such check would be redundant.

Reply:
Added `returnvalue` check

Reply:
Seems that problem should be covered